tests/test_ci_self_hosted_guard.sh

```
@@ -174,6 +174,111 @@ check_release_helper_upload_retry() {
+    ' "$file"; then
+      echo "FAIL: $(basename "$file") must install sentry-cli through scripts/ensure-sentry-cli.sh before dSYM upload"
+      exit 1
+    fi
+  done
+
+  echo "PASS: dSYM upload installs sentry-cli without requiring Homebrew"
+}
+
+check_dmg_signing_uses_build_keychain() {
+  for file in "$ROOT_DIR/.github/workflows/nightly.yml" "$ROOT_DIR/.github/workflows/release.yml"; do
+    if grep -Fq -- '--identity="$APPLE_SIGNING_IDENTITY"' "$file"; then
```

Shellcheck (0.11.0)

[info] Expressions don't expand in single quotes, use double quotes for that.

(SC2016)